### PR TITLE
Implemented Reset Networking for sACN Receiver

### PR DESCRIPTION
**Pull request by ChristianReese**
Here's the Reset Networking implementation for the sACN Receiver.

### CODE REVIEW

File: src/sacn/receiver.c [hunk 1]
```diff
@@ -845,6 +973,24 @@ void remove_receiver_sockets(SacnReceiver* receiver, bool close_now)
...
 }
 
+/*
+ * Remove all receiver sockets, choosing whether to close them now or wait until the next thread cycle.
+ *
+ * The sACN lock should be locked before calling this.
+ *
+ * [in] close_now Whether to close the sockets now or wait until the next thread cycle.
+ */
+void remove_all_receiver_sockets(bool close_now)
```
Comment (samkearney) [inline src/sacn/receiver.c:983]: I know that the remove_receiver_sockets() function was there before, but I'm not a fan of boolean flag parameters for stuff like this, they make it hard to read what's happening at the call site of the function:
```c
remove_all_receiver_sockets(true); // What does true mean??
```

If it's not too high impact, consider changing to an enum:
```c
typedef enum
{
  kCloseSocketNow,
  kQueueSocketForClose
} socket_close_behavior_t;

void remove_all_receiver_sockets(socket_close_behavior_t close_behavior)
```

**Review by samkearney — APPROVED** [review]

File: src/sacn/receiver.c (hunk 1 above)
Comment (ChristianReese) [reply src/sacn/receiver.c:983]: Would you say there are still cases when boolean flag parameters should be used? If so what are those cases?

Comment (samkearney) [reply src/sacn/receiver.c:983]: They are appropriate when the function name gives you enough information to infer the parameter's usage from context. e.g.:

```c
set_rdm_enabled(true);
```

Comment (ChristianReese) [reply src/sacn/receiver.c:983]: Makes sense - in all cases, readability at the call site is key, so if it isn't readable with a bool, switch to an enum. I'll go ahead and make this change. Thanks!

Comment (nwagner) [reply src/sacn/receiver.c:983]: Good catch & discussion!

**Review by nwagner — APPROVED** [review]

**Review by samkearney — APPROVED** [review]

**Pull request merged**